src/Service/MailTemplate/ComplexMacroHelper.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace App\Service\MailTemplate;
+
+use App\Entity\Agenda;
+use App\Entity\BoardRole;
+use App\Entity\CaseEntity;
+use PhpOffice\PhpWord\Element\Link;
+use PhpOffice\PhpWord\Element\Row;
+use PhpOffice\PhpWord\Element\Table;
+use PhpOffice\PhpWord\SimpleType\Jc;
+use PhpOffice\PhpWord\SimpleType\TblWidth;
+use Symfony\Component\Routing\RouterInterface;
+
+class ComplexMacroHelper
+{
+    public function __construct(private RouterInterface $router)
+    {
+    }
+
+    /**
+     * @param $entity
+     *
+     * @return array|ComplexMacro[]
+     */
+    public function buildMacros($entity): array
+    {
+        $values = [];
+
+        if ($entity instanceof CaseEntity) {
+            // Note: Setting text on the link will break the link.
+            $values['case.link'] = new ComplexMacro(
+                new Link($this->router->generate('case_show', ['id' => $entity->getId()], RouterInterface::ABSOLUTE_URL)),
+                'Url with link to the case'
+            );
```

Comment:
What is the use case of this? Was it just an intermediate step before adding the agenda link?